content/3_Silver/Prefix_Sums.mdx

```diff
@@ -681,6 +692,305 @@ as expected.
 
 Since no matter the size of the submatrix we are summing, we only need to access four values of the 2D prefix sum array, this runs in $\mathcal{O}(1)$ per query after an $\mathcal{O}(NM)$ preprocessing.
 
+## Creation From Endpoints
+
+<FocusProblem problem={problems.sample3[0]} />
+
+### 1-Dimensional
+
+A common application of prefix sums is to generate a prefix sum array from given interval endpoints in the form $[a, b]$
+
+We are given the problem that for many such intervals, compute the point with the most overlapping intervals. 
+
+Naïvely, we can compute the interval by looping through each indice from $[a,b]$ individually, resulting in an $\mathcal{O}(NQ)$ time complexity (where N is the maximum size of our interval and Q is the number of queries), which easily times out (think what happens when the interval [0, N] is queried many times).
+
+We can do better. It's easy to see that an increment of $x$ (before computation) in $\texttt{prefix}[i]$ causes all subsequent $\texttt{prefix}[i...N]$ (after computation) to increase by $x$. This concept can be conceptualized through increment and decrement points. An increment point increases (and decrement decreases) all subsequent cells. (Note that our decrement point is located at $B+1$ because the interval is inclusive.)
+
+Example: Add two to each point in the interval [2, 4]
+
+Our matrix after adding 2 at our increment (start) point (before computation)
+<center>
+	<table className='text-center'>
+		<tr>
+			<td>0</td>
+			<td>0</td>
+			<td className='bg-green-200'>2</td>
+			<td>0</td>
+			<td>0</td>
+			<td>0</td>
+		</tr>
+	</table>
+</center>
+
+Our matrix after adding 2 at our increment (start) point (and computing).
+<center>
+	<table className='text-center'>
+		<tr>
+			<td>0</td>
+			<td>0</td>
+			<td className='bg-green-200'>2</td>
+			<td>2</td>
+			<td>2</td>
+			<td>2</td>
+		</tr>
+	</table>
+</center>
+
+Our matrix after subtracting 2 at our decrement point (and computing).
+<center>
+	<table className='text-center'>
+		<tr>
+			<td>0</td>
+			<td>0</td>
+			<td>2</td>
+			<td>2</td>
+			<td>2</td>
+			<td className='bg-red-200'>0</td>
+		</tr>
+	</table>
+</center>
+
+In this above problem, we are only incrementing by 1, so our $x$ is consistent. As a result, when we encounter a starting point, we can increment by 1, and for an endpoint, decrement by 1. For our problem, we don't need to include a prefix array because we aren't querying a range, so a single variable suffices.
+
+### Restaurant Customers Solution
```

Comment:
Can we have 2 solutions? This (alternative) short/faster approach and also a main approach that uses coord compress to explicitly compute the prefix sum array.

Reply:
Scratch that. It would be helpful, but not necessary if you'd rather not

Reply:
lol I coded it already I think it's in the new commit